pkg/apis/serving/v1beta1/predictor.go

```diff
@@ -101,6 +101,11 @@ func (s *PredictorSpec) GetImplementation() ComponentImplementation {
 	return s.GetImplementations()[0]
 }
 
+// IsEmpty returns if all components does not exist
+func (s *PredictorSpec) IsEmpty() bool {
```

Comment:
How this is different from `GetPredictor` ?

Reply:
Hmm I may need to refractor the names for clarity. `GetPredictor` will return a `PredictorImplementation` if one exists. I wanted a way to retrieve a `ComponentImplementation` if one exists and I decided to create a function `IsEmpty` to determine if any component implementation exists.